depends/common/libjpeg-turbo/0004-fix-use-of-CMAKE_SYSTEM_PROCESSOR.patch

```diff
@@ -0,0 +1,16 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -40,7 +40,11 @@
+ 
+ # Detect CPU type and whether we're building 64-bit or 32-bit code
+ math(EXPR BITS "${CMAKE_SIZEOF_VOID_P} * 8")
+-string(TOLOWER ${CMAKE_SYSTEM_PROCESSOR} CMAKE_SYSTEM_PROCESSOR_LC)
++if(CMAKE_SYSTEM_PROCESSOR)
++  string(TOLOWER ${CMAKE_SYSTEM_PROCESSOR} CMAKE_SYSTEM_PROCESSOR_LC)
++elseif(CPU)
++  string(TOLOWER ${CPU} CMAKE_SYSTEM_PROCESSOR_LC)
```

Comment:
No hacks, set CMAKE_SYSTEM_PROCESSOR in the toolchain.

Reply:
I look about a fix inside Kodi, it was not reproducible at home, thats why I added this hack. 

Also for things like them, I add you as reviewer, but please write with a bit nicer words :pensive:!

Reply:
For which platform did you try to reproduce?

Reply:
Has tested on Linux and my Mac, for a unknown reason was there the `CMAKE_SYSTEM_PROCESSOR` available but by Jenkins not.

Here a old build about, without them https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Fimagedecoder.mpo/detail/PR-8/2/pipeline

Reply:
You are building the add-on the wrong way?

Reply:
I start today from here https://github.com/xbmc/xbmc/blob/master/tools/depends/xbmc-addons.include to find the reason why this not becomes given.

Reply:
On OSX I use normally the small step like on Linux, but also for cases like them the same way like Jenkins over the make.

```sh
make -j $BUILDTHREADS ADDONS='${addon}' ADDONS_DEFINITION_DIR=`pwd`/addons ADDON_SRC_PREFIX=`pwd` EXTRA_CMAKE_ARGS="-DPACKAGE_ZIP=ON -DPACKAGE_DIR=`pwd`/../../../../cmake/addons/build/zips" PACKAGE=1
```

Reply:
Have created here a request about: https://github.com/xbmc/xbmc/pull/17363

To become usable for here then also a backport to Leia is needed.

Reply:
@Rechi would relate it to Leia, wouldn't it be better to do it here with the Hacky Patch? And at Matrix a revised build system and omit this hack?

To make this suitable for Leia on Kodi would be a little much, with things already released.

Reply:
No hacks. If you can't fix it the right way, just keep the current library version, which works fine.

Reply:
I look then also for backward to fix this https://github.com/xbmc/imagedecoder.raw/issues/20
For here OK to take the old but the https://github.com/xbmc/imagedecoder.raw/pull/22 contains the same.